File: backend/utils/image_utils.py
import os
from pathlib import Path

def get_image_url(image_path: str) -> str:
    """
    Convert stored image path to relative path for frontend to construct full URL

    Args:
        image_path: Relative path like '/uploads/image.jpg' or 'uploads/image.jpg'

    Returns:
        Relative path starting with /
    """
    if not image_path:
        return None

    # If already a full URL, return as-is
    if image_path.startswith('http://') or image_path.startswith('https://'):
        return image_path

    # Ensure path starts with /
    if not image_path.startswith('/'):
        image_path = '/' + image_path

    return image_path


def save_image_path(file_path: str) -> str:
    """
    Save image path in a consistent format (relative path)

    Args:
        file_path: Full file path like '/uploads/image.jpg'

    Returns:
        Relative path suitable for storage
    """
    if not file_path:
        return None

    # Remove any base URL if present
    if file_path.startswith('http://') or file_path.startswith('https://'):
        # Extract just the path portion
        from urllib.parse import urlparse
        parsed = urlparse(file_path)
        file_path = parsed.path

    # Ensure it starts with /
    if not file_path.startswith('/'):
        file_path = '/' + file_path

    return file_path

File: frontend/src/utils/imageUtils.js

import { API_BASE_URL } from '../api/client';

/**
 * Convert relative image path to full URL
 * @param {string} imagePath - Relative path like '/uploads/image.jpg'
 * @returns {string} Full URL for the image
 */
export const getFullImageUrl = (imagePath) => {
  if (!imagePath) {
    return null;
  }

  // If already a full URL, return as-is
  if (imagePath.startsWith('http://') || imagePath.startsWith('https://')) {
    return imagePath;
  }

  // Ensure path starts with /
  const path = imagePath.startsWith('/') ? imagePath : `/${imagePath}`;

  // Construct full URL using API_BASE_URL
  return `${API_BASE_URL}${path}`;
};

/**
 * Convert array of relative image paths to full URLs
 * @param {string[]} imagePaths - Array of relative paths
 * @returns {string[]} Array of full URLs
 */
export const getFullImageUrls = (imagePaths) => {
  if (!imagePaths || !Array.isArray(imagePaths)) {
    return [];
  }
  return imagePaths.map(getFullImageUrl).filter(Boolean);
};
